homeassistant/components/recorder/db_schema.py

```diff
@@ -298,18 +300,19 @@ def _time_fired_isotime(self) -> str | None:
     @staticmethod
     def from_event(event: Event) -> Events:
         """Create an event database object from a native event."""
+        context = event.context
         return Events(
             event_type=None,
             event_data=None,
-            origin_idx=EVENT_ORIGIN_TO_IDX.get(event.origin),
+            origin_idx=EVENT_ORIGIN_VALUE_TO_IDX[event.origin.value],
```

Comment:
Using .value avoids calling the enum `__hash__` method and instead uses the native code `str` hash